taskmanager/taskmanager.go

```diff
@@ -1,4 +1,161 @@
 /*
-Package taskmanager is the task/job management service layer for concurrent and asynchronous tasks
+Package taskmanager is the task/job management service layer for concurrent and asynchronous tasks with cron scheduling.
 */
 package taskmanager
+
+import (
+	"context"
+
+	"github.com/BuxOrg/bux/logging"
+	"github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/robfig/cron/v3"
+	"github.com/rs/zerolog"
+	taskq "github.com/vmihailenco/taskq/v3"
+)
+
+type (
+
+	// TaskManager implements the Tasker interface
+	TaskManager struct {
+		options *options
+	}
+
+	// options holds all the configuration for the client
+	options struct {
+		cronService     *cron.Cron      // Internal cron job client
+		debug           bool            // For extra logs and additional debug information
+		logger          *zerolog.Logger // Internal logging
+		newRelicEnabled bool            // If NewRelic is enabled (parent application)
+		taskq           *taskqOptions   // All configuration and options for using TaskQ
+	}
+
+	// taskqOptions holds all the configuration for the TaskQ engine
+	taskqOptions struct {
+		config *taskq.QueueOptions    // Configuration for the TaskQ engine
+		queue  taskq.Queue            // Queue for TaskQ
+		tasks  map[string]*taskq.Task // Registered tasks
+	}
+)
+
+// NewTaskManager creates a new client for all TaskManager functionality
+// If no options are given, it will use local memory for the queue.
+// ctx may contain a NewRelic txn (or one will be created)
+func NewTaskManager(ctx context.Context, opts ...ClientOps) (Tasker, error) {
+	// Create a new tm with defaults
+	tm := &TaskManager{options: &options{
+		debug:           false,
+		newRelicEnabled: false,
+		taskq: &taskqOptions{
+			tasks:  make(map[string]*taskq.Task),
+			config: DefaultTaskQConfig("taskq"),
+		},
+	}}
+
+	// Overwrite defaults with any set by user
+	for _, opt := range opts {
+		opt(tm.options)
+	}
+
+	// Set logger if not set
+	if tm.options.logger == nil {
+		tm.options.logger = logging.GetDefaultLogger()
+	}
+
+	// Use NewRelic if it's enabled (use existing txn if found on ctx)
+	// ctx = tm.options.getTxnCtx(ctx)
```

Comment:
why this is commented out?

Reply:
This had been commented out before my changes so I left this as is. I don't know why. But I'm not familiar with the `NewRelic` feature.

Reply:
ok, I didn't realize that it was commented out previously